src/auth/authclient.ts

```diff
@@ -317,11 +325,22 @@ export abstract class AuthClient
     ) {
       headers.set('x-goog-user-project', this.quotaProjectId);
     }
+    if (
+      !headers.has('x-goog-allowed-locations') &&
```

Comment:
I think we need to send the most up to date TB data. The `!headers.has('x-goog-allowed-locations')` check will skip setting and updating the value it if was previously set.